Test BC for built-in torchbind methods

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38571 Remove BC hack
* **#38560 Test BC for built-in torchbind methods**

Schema changes will cause breakage that looks like:

```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
        __getstate__(__torch__.torch.classes.rnn.CellParamsBase _0) -> ((str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _0)
        __setstate__(__torch__.torch.classes.rnn.CellParamsBase _0, (str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _1) -> (None _0)
]

```

Differential Revision: [D21598067](https://our.internmc.facebook.com/intern/diff/D21598067)

Commit:
Test BC for built-in torchbind methods

[ghstack-poisoned]

Commit:
Update on "Test BC for built-in torchbind methods"


Schema changes will cause breakage that looks like:

```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
        __getstate__(__torch__.torch.classes.rnn.CellParamsBase _0) -> ((str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _0)
        __setstate__(__torch__.torch.classes.rnn.CellParamsBase _0, (str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _1) -> (None _0)
]

```

Differential Revision: [D21598067](https://our.internmc.facebook.com/intern/diff/D21598067)

[ghstack-poisoned]

zdevito commented on torch/custom_class_detail.h:
```diff
@@ -148,6 +148,10 @@ TORCH_API at::ClassTypePtr getCustomClass(const std::string& name);
 
+// This API is for testing purposes ONLY. It should not be used in
+// any load-bearing code.
+TORCH_API std::vector<std::unique_ptr<jit::Function>>& customClassMethods();
```
This should return the minimum necessary, which is just the schemas not the functions themselves. It should also have the language that it is for debug only in the function name so it will get caught in review `schemaOfCustomClassesForBCCheck()`

Commit:
Update on "Test BC for built-in torchbind methods"


Schema changes will cause breakage that looks like:

```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
        __getstate__(__torch__.torch.classes.rnn.CellParamsBase _0) -> ((str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _0)
        __setstate__(__torch__.torch.classes.rnn.CellParamsBase _0, (str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _1) -> (None _0)
]

```

Differential Revision: [D21598067](https://our.internmc.facebook.com/intern/diff/D21598067)

[ghstack-poisoned]

Commit:
Update on "Test BC for built-in torchbind methods"


Schema changes will cause breakage that looks like:

```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
        __getstate__(__torch__.torch.classes.rnn.CellParamsBase _0) -> ((str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _0)
        __setstate__(__torch__.torch.classes.rnn.CellParamsBase _0, (str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _1) -> (None _0)
]

```

Differential Revision: [D21598067](https://our.internmc.facebook.com/intern/diff/D21598067)

[ghstack-poisoned]

Commit:
Update on "Test BC for built-in torchbind methods"


Schema changes will cause breakage that looks like:

```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
        __getstate__(__torch__.torch.classes.rnn.CellParamsBase _0) -> ((str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _0)
        __setstate__(__torch__.torch.classes.rnn.CellParamsBase _0, (str, Tensor[], float[], int[], __torch__.torch.classes.quantized.LinearPackedParamsBase[], int) _1) -> (None _0)
]

```

Differential Revision: [D21598067](https://our.internmc.facebook.com/intern/diff/D21598067)

[ghstack-poisoned]
